tests/data_manager/test_ticker_manager.py

```diff
@@ -0,0 +1,94 @@
+from tests.util.random_numbers import random_price, random_quantity, random_timestamp
+
+pytestmark = pytest.mark.asyncio
+
+
+@pytest.fixture()
+async def ticker_manager():
+    t_manager = TickerManager()
+    await t_manager.initialize()
+    return t_manager
+
+
+async def test_init(ticker_manager):
+    assert not ticker_manager.ticker == {}
+    assert not ticker_manager.mini_ticker == {}
+
+
+async def test_update_and_reset_ticker(ticker_manager):
+    price = random_price()
+    tm = random_timestamp()
+    vlm = random_quantity()
+    ticker_manager.ticker_update({
+        ExchangeConstantsTickersColumns.ASK.value: price,
+        ExchangeConstantsTickersColumns.ASK_VOLUME.value: vlm,
+        ExchangeConstantsTickersColumns.TIMESTAMP.value: tm
+    })
+    assert ticker_manager.ticker[ExchangeConstantsTickersColumns.ASK.value] == price
+    assert ticker_manager.ticker[ExchangeConstantsTickersColumns.ASK_VOLUME.value] == vlm
+    assert ticker_manager.ticker[ExchangeConstantsTickersColumns.TIMESTAMP.value] == tm
+    if not os.getenv('CYTHON_TEST_IGNORE'):
+        ticker_manager.reset_ticker()
+        assert ticker_manager.ticker == {
+            ExchangeConstantsTickersColumns.ASK.value: nan,
+            ExchangeConstantsTickersColumns.ASK_VOLUME.value: nan,
+            ExchangeConstantsTickersColumns.BID.value: nan,
+            ExchangeConstantsTickersColumns.BID_VOLUME.value: nan,
+            ExchangeConstantsTickersColumns.OPEN.value: nan,
+            ExchangeConstantsTickersColumns.LOW.value: nan,
+            ExchangeConstantsTickersColumns.HIGH.value: nan,
+            ExchangeConstantsTickersColumns.CLOSE.value: nan,
+            ExchangeConstantsTickersColumns.LAST.value: nan,
+            ExchangeConstantsTickersColumns.AVERAGE.value: nan,
+            ExchangeConstantsTickersColumns.SYMBOL.value: nan,
+            ExchangeConstantsTickersColumns.QUOTE_VOLUME.value: nan,
+            ExchangeConstantsTickersColumns.TIMESTAMP.value: 0,
+            ExchangeConstantsTickersColumns.VWAP.value: nan
+        }
+
+
+async def test_update_and_reset_mini_ticker(ticker_manager):
+    vol = random_quantity()
+    open_p = random_price()
+    ticker_manager.mini_ticker_update({
+            ExchangeConstantsMiniTickerColumns.HIGH_PRICE.value: random_price(),
+            ExchangeConstantsMiniTickerColumns.LOW_PRICE.value: random_price(),
+            ExchangeConstantsMiniTickerColumns.OPEN_PRICE.value: open_p,
+            ExchangeConstantsMiniTickerColumns.CLOSE_PRICE.value: random_price(),
+            ExchangeConstantsMiniTickerColumns.VOLUME.value: vol,
+            ExchangeConstantsMiniTickerColumns.TIMESTAMP.value: random_timestamp()
+        })
+    assert ticker_manager.mini_ticker[ExchangeConstantsMiniTickerColumns.VOLUME.value] == vol
+    assert ticker_manager.mini_ticker[ExchangeConstantsMiniTickerColumns.OPEN_PRICE.value] == open_p
+    if not os.getenv('CYTHON_TEST_IGNORE'):
```

Comment:
sad that we can't test this on compiled env :(

Reply:
Yes it's because we need to use private method to test it :(